crates/tauri/src/window/mod.rs

```diff
@@ -1989,6 +1990,36 @@ tauri::Builder::default()
       .set_title_bar_style(style)
       .map_err(Into::into)
   }
+
+  /// Sets the theme for this window.
+  ///
+  /// ## Platform-specific
+  ///
+  /// - **Linux / macOS**: Theme is app-wide and not specific to this window.
+  /// - **iOS / Android:** Unsupported.
+  pub fn set_theme(&self, theme: Option<Theme>) -> crate::Result<()> {
+    self
+      .window
+      .dispatcher
+      .set_theme(theme)
+      .map_err(Into::<crate::Error>::into)?;
+    #[cfg(windows)]
+    if let (Some(window_menu), Ok(hwnd)) = (&*self.menu_lock(), self.hwnd()) {
+      let raw_hwnd = hwnd.0 as isize;
+      let menu = window_menu.menu.clone();
```

Comment:
instead of holding the `menu_lock` here for the rest of the scope, we can just clone and release the lock
```suggestion
    let window_menu = self.menu_lock().clone();
    if let (Some(menu), Ok(hwnd)) = (window_menu, self.hwnd()) {
      let raw_hwnd = hwnd.0 as isize;
```